ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala

```diff
@@ -1120,44 +1137,53 @@ abstract class AbstractHttpServiceIntegrationTest
     }
   }
 
-  "should be able to serialize and deserialize domain commands" in withLedger { client =>
-    instanceUUIDLogCtx(implicit lc =>
-      jsonCodecs(client).map { case (encoder, decoder) =>
-        testCreateCommandEncodingDecoding(encoder, decoder)
-        testExerciseCommandEncodingDecoding(encoder, decoder)
-      }: Future[Assertion]
-    )
+  "should be able to serialize and deserialize domain commands" in withLedger {
+    (client, ledgerId) =>
+      instanceUUIDLogCtx(implicit lc =>
+        jsonCodecs(client, ledgerId).flatMap { case (encoder, decoder) =>
+          testCreateCommandEncodingDecoding(encoder, decoder, ledgerId) *>
+            testExerciseCommandEncodingDecoding(
+              encoder,
+              decoder,
+              ledgerId,
+            )
+        }: Future[Assertion]
+      )
   }
 
   private def testCreateCommandEncodingDecoding(
       encoder: DomainJsonEncoder,
       decoder: DomainJsonDecoder,
-  ): Assertion = {
+      ledgerId: LedgerId,
+  ): Future[Assertion] = instanceUUIDLogCtx { implicit lc =>
     import json.JsonProtocol._
     import util.ErrorOps._
 
     val command0: domain.CreateCommand[v.Record, OptionalPkg] = iouCreateCommand()
 
-    val x = for {
-      jsVal <- encoder.encodeCreateCommand(command0).liftErr(JsonError)
-      command1 <- decoder.decodeCreateCommand(jsVal)
+    val x: EitherT[Future, JsonError, Assertion] = for {
+      jsVal <- EitherT.either[Future, JsonError, JsValue](
+        encoder.encodeCreateCommand(command0).liftErr(JsonError)
+      )
```

Comment:
Ascription is more declarative than tparam-passing.

```suggestion
    type F[A] = EitherT[Future, JsonError, A]
    val x: F[Assertion] = for {
      jsVal <- EitherT.either(
        encoder.encodeCreateCommand(command0).liftErr(JsonError)
      ): F[JsValue]
```